source/extensions/filters/http/fault/fault_filter.cc

```diff
@@ -129,34 +132,35 @@ Http::FilterHeadersStatus FaultFilter::decodeHeaders(Http::HeaderMap& headers, b
   return Http::FilterHeadersStatus::Continue;
 }
 
+bool FaultFilter::faultOverflow() {
+  const uint64_t max_faults = config_->runtime().snapshot().getInteger(
+      RuntimeKeys::get().MaxActiveFaultsKey, fault_settings_->maxActiveFaults().has_value()
+                                                 ? fault_settings_->maxActiveFaults().value()
+                                                 : std::numeric_limits<uint64_t>::max());
+  if (config_->stats().active_faults_.value() >= max_faults) {
```

Comment:
These stats are per thread right or per HCM instance pre aggregation? my knowledge of where stats get aggregated is not that great.

If its per thread, (a) docs need some nuancing to convey that this is a soft threshold per thread and should not be thought of as a global limit for this virtual host.. (b) this has fidelity issues where one HCM on a port could have exceeded the threshold, while another one on a different port is unable to inject faults any more.

If its per HCM instance, i.e. one per connection (H1/H2), then docs need changes accordingly;

Reply:
It's global for an individual HTTP listener stat name (multiple listeners can have the same stat name). Given this do you still think the docs need updating?

Reply:
@mattklein123 do we need to use atomic compare-and-exchange to track active_faults_? I wonder if we have a use-after-check issue here where we first check and then increase?

Reply:
Yeah in the current implementation it can go over slightly, but this is similar to how circuit breakers also work currently, so I don't think it's a big deal. I will add comment/doc.